components/inputs/input-date.js

```diff
@@ -211,12 +225,46 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		super.updated(changedProperties);
 
 		changedProperties.forEach((oldVal, prop) => {
-			if (prop === 'value') {
-				this._formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : '';
+			if (prop === '_dateTimeDescriptor' || prop === 'value') {
+				this._setFormattedValue();
 			}
 		});
 	}
 
+	_getContentWidth() {
+		const text = document.createElement('div');
+		document.body.appendChild(text);
```

Comment:
Nit: you may want to append it after setting all the style properties, just in case changing the style after it's in the DOM would cause a repaint.